docs/pages/management/guides/joining-nodes-kubernetes-serviceaccount.mdx

```diff
@@ -0,0 +1,242 @@
+  join_method: kubernetes
+
+  kubernetes:
+    allow:
+      # Service account names follow the format "namespace:serviceaccountname"
+      - service_account: "teleport-agent:teleport-app-node"
+```
+
+<Tabs>
+<TabItem label="Using local `tctl` as your current user">
+Make sure your local `tctl` is at least at version 12 with `tctl version`.
+
+Run `tctl create token.yaml` to create the token.
+
+Finally, validate the token was created:
+
+```code
+$ tctl get token/kubernetes-token
+
+kind: token
+metadata:
+  expires: "3000-01-01T00:00:00Z"
+  name: kubernetes-token
+spec:
+  join_method: kubernetes
+  roles:
+  - App
+version: v2
+```
+
+</TabItem>
+<TabItem label="Using remote `tctl` as admin">
+Retrieve the name and namespace of the Auth Nodes deployment:
+
+```code
+$ kubectl get namespaces
+NAME              STATUS   AGE
+cert-manager      Active   40d
+default           Active   40d
+kube-system       Active   40d
+teleport          Active   40d
+
+# We look for deployments in the "teleport" namespace
+$ kubectl get deployments -n teleport
+NAME             READY   UP-TO-DATE   AVAILABLE   AGE
+teleport-auth    2/2     2            2           6d20h
+teleport-proxy   2/2     2            2           6d20h
+
+# Here, the deployment name is "teleport-auth".
+```
+
+Then run the following command to execute the `tctl create` directly from inside
+one of the auth pod:
+
+```code
+$ kubectl exec -i -n teleport deployment/teleport-auth -- tctl create < token.yaml
+```
+
+Finally, validate the token was successfully created:
+
+```code
+$ kubectl exec -i -n teleport deployment/teleport-auth tctl get token/kubernetes-token
+
+kind: token
+metadata:
+  expires: "3000-01-01T00:00:00Z"
+  name: kubernetes-token
+spec:
+  join_method: kubernetes
+  roles:
+  - App
+version: v2
+```
+
+</TabItem>
+</Tabs>
+
+## Step 2/5. Deploy a demonstration HTTP app
+
+In this step we deploy a demonstration HTTP application and don't expose it
+publicly. This application will be used to validate the Teleport Application
+Access Node works as expected.
+
+```code
+$ kubectl create namespace demo-app
+namespace/demo-app created
+
+$ kubectl create deployment --image=nginx --namespace demo-app --port=80 demo-app
+deployment.apps/demo-app created
+
+$ kubectl expose deployment demo-app -n demo-app --port=80 --target-port=80 --selector='app=demo-app'
+service/demo-app exposed
+```
+
+Validate the application pods are running and ready with the following command:
+
+```code
+$ kubectl get pods -n demo-app
+NAME                        READY   STATUS    RESTARTS   AGE
+demo-app-7664d59cb8-bvbmz   1/1     Running   0          67s
+```
+
+## Step 3/5. Configure your Application Access Nodes
+
+The Kubernetes join method can be used for Teleport services running SSH, Proxy,
+Kubernetes, Application, Database, or Windows desktop roles. The service should
+be run in the same Kubernetes cluster and have a Kubernetes ServiceAccount Token
+mounted. [The `teleport-kube-agent`
```

Comment:
I think we should make it clearer that this guide will show you how to install the `teleport-kube-agent` chart so readers don't go looking for how to install it when they read this. Maybe we could move the information about the ServiceAccount Token from this paragraph to Step 4, when we install the Helm chart?